rosbag2_transport/include/rosbag2_transport/player.hpp

```diff
@@ -201,6 +245,7 @@ class Player : public rclcpp::Node
   void play_messages_from_queue();
   void prepare_publishers();
   bool publish_message(rosbag2_storage::SerializedBagMessageSharedPtr message);
+  static callback_handle_t get_new_on_play_msg_callback_handle();
```

Comment:
It is static because it has static atomic variable inside 
```cpp
Player::callback_handle_t Player::get_new_on_play_msg_callback_handle()
{
  static std::atomic<callback_handle_t> handle_count{0};
  return handle_count.fetch_add(1, std::memory_order_relaxed) + 1;
}
```
This function implements lock-free (without mutex lock) sort of new `unique ID` generation for multi-threaded environment.  
This function needs to be `static`. Declaring it as static will share this method between instantiated instances of the `Player` class to be able to provide uniqueID even between different instances.  
For example we can have 
```cpp
Player player1;
Player player2;
auto callback_handle1 = player1.add_on_play_message_post_callback(function1);
auto callback_handle2 = player2.add_on_play_message_post_callback(function1);
```
Declaring it static give us guarantee that callback handles will be unique even for `player1` and `player2`.

It looks pretty simple, however solving complex problem for unique ID generation in a very efficient way.